libretroshare/src/retroshare/rsgrouter.h

```diff
@@ -19,19 +20,29 @@
  * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
  *                                                                             *
  *******************************************************************************/
-
 #pragma once
 
 #include "util/rsdir.h"
+#include "util/rsdeprecate.h"
 #include "retroshare/rsids.h"
 #include "retroshare/rsgxsifacetypes.h"
+#include "rsitems/rsserviceids.h"
+
+RS_DEPRECATED_FOR(RsGxsId)
+typedef RsGxsId GRouterKeyId;
+
+/**
+ * @note Some items use this type, when migrating to RsServiceType (2 bytes),
+ * special care should be taken to do it in a way which doesn't break
+ * retrocompatibility
+ */
+RS_DEPRECATED_FOR(RsServiceType)
```

Comment:
I dont understand that. The GRouterServiceId is rather a client ID. Such a client is not necessarily a RSService. So merging the two limit possibilities and does not help much either.